### PR TITLE
KEYCLOAK-18147 Fix CodeQL failures

### PR DESCRIPTION
**Pull request by hmlnarik**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -32,6 +36,10 @@ jobs:
     - name: Checkout repository
       uses: actions/checkout@v2
 
+    - uses: whelk-io/maven-settings-xml-action@v15
```
Comment (stianst) [inline .github/workflows/codeql-analysis.yml:39]: Did you review this action? Custom actions needs to be thoroughly reviewed to make sure they are not doing something they shouldn't

Comment (hmlnarik) [reply .github/workflows/codeql-analysis.yml:39]: Good point. I took the same approach as in #8007.
@vmuzikar Have you reviewed this action in detail?

Comment (vmuzikar) [reply .github/workflows/codeql-analysis.yml:39]: I haven't done a line-by-line review, but the action seems pretty simple and straightforward. First, it [prepares in-memory maven xml](https://github.com/whelk-io/maven-settings-xml-action/blob/v15/src/settings.js) based on a [predefined template](https://github.com/whelk-io/maven-settings-xml-action/blob/v15/template/settings.xml), and then it [writes it to the .m2 dir](https://github.com/whelk-io/maven-settings-xml-action/blob/v15/src/index.js). Doesn't seem dangerous.

Comment (stianst) [reply .github/workflows/codeql-analysis.yml:39]: What the action says it does and what it does can be two very different things, and the action could also be compromised down the road.

Recommendations here are to use SHA and not version as a version can be modified. Also, to carefully review all actions that are not published by an already trusted source like GitHub. We haven't done this properly in the past, but should start doing it. Usually an action is only a few lines of code, so doesn't take much time to scan through.

**Comment by hmlnarik** [comment]
Model tests failed due to [KEYCLOAK-16999](https://issues.redhat.com/browse/KEYCLOAK-16999).

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (hmlnarik) [reply .github/workflows/codeql-analysis.yml]: I took a different approach without a need for external actions. Could you please re-review?

**Review by vmuzikar — CHANGES_REQUESTED** [review]
The QS tests seem to be broken by this change. That is because the `settings.xml` file is missing from the QS repo. The way I see it there are two ways how to fix it:
1. Introduce similar change to the QS repo. The `maven-settings-xml-action` is used there as well so we might want to get rid of it there as well.
2. Use `settings.xml` from the KC repo instead of QS repo, i.e. `keycloak/.github/settings.xml`.

I'm more inclined towards 1. as we should use the same approach for both KC and QS repo. Also, in case we'd need to use some QS-specific config in `settings.xml` in the QS repo, it'd be cleaner to use the file from there and not from the KC repo.

**Comment by hmlnarik** [comment]
Related QS update: keycloak/keycloak-quickstarts#252

**Comment by hmlnarik** [comment]
@vmuzikar Could you please re-review?
The `CIBATest` failure is apparently fixed in master already in #8213.

**Review by vmuzikar — APPROVED** [review]
@hmlnarik LGTM, thanks!

**Pull request merged**